core/src/main/scala/kafka/log/Log.scala

```diff
@@ -262,12 +261,14 @@ class Log(val dir: File,
                                      initFileSize = this.initFileSize(),
                                      preallocate = config.preallocate))
     } else {
-      recoverLog()
-      // reset the index size of the currently active log segment to allow more entries
-      activeSegment.index.resize(config.maxIndexSize)
-      activeSegment.timeIndex.resize(config.maxIndexSize)
+      if (!dir.getAbsolutePath.endsWith(Log.DeleteDirSuffix)) {
```

Comment:
If a particular broker holds a large amount of data for a particular topic or if we have a high throughput topic that was marked for delete, then deleting them synchronously when the broker starts up might increase the startup time significantly.